### PR TITLE
cli: Initial runtime cli implementation. 

### PR DESCRIPTION
**Pull request by jcvenegas**
This PR adds cli implementation based on kata-virtcontainers.

Add kata-runtime
Add unit test
Add Makefile to build cli
Remove runv and cc-runtime git submodules
Move vendor to root project.

Signed-off-by: Julio Montes julio.montes@intel.com
Signed-off-by: James O. D. Hunt james.o.hunt@intel.com
Signed-off-by: Jose Carlos Venegas Munoz jose.carlos.venegas.munoz@intel.com

### CODE REVIEW

**Comment by devimc** [comment]
lgtm

**Comment by sboeuf** [comment]
lgtm

**Comment by jcvenegas** [comment]
branch updated.

- Modified travis scripts to build cli
- Modified main Makefile to build cli

File: Gopkg.lock [hunk 1]
```diff
@@ -31,12 +37,24 @@
   packages = ["pkg/ns"]
   revision = "7f98c94613021d8b57acfa1a2f0c8d0f6fd7ae5a"
 
+[[projects]]
+  name = "github.com/containers/virtcontainers"
```
Comment (bergwolf) [inline Gopkg.lock]: I guess we can just remove this now that virtcontainers lives in kata-containers/runtime/?

File: cli/VERSION [hunk 2]
```diff
@@ -0,0 +1 @@
+1.0.0
```
Comment (bergwolf) [inline cli/VERSION]: Are we starting from v1.0.0 release? I guess it should be something like v0.1 at the moment?

**Comment by bergwolf** [comment]
It seems the commit that `Remove runv and cc-runtime git submodules` is missing?

File: Gopkg.lock (hunk 1 above)
Comment (jcvenegas) [reply Gopkg.lock]: This is auto-generated by `dep ensure`, Somehow is still resolving to the old repo as dependency. I'll check why.

Comment (jodh-intel) [reply Gopkg.lock]: I think it's due to the `cc-runtime` sub-module.

**Comment by jcvenegas** [comment]
@bergwolf @jodh-intel, I think git submodule could be removed in a next PR. I got some issue with CI trying to remove it.  IMO is better to handle that after merge the CLI implementation. 

This PR is almost ready, we only have a few CPU test to fix.  I can try to split this PR in 2 cli commands and unit test if that helps to reduce the size of this PR.

**Comment by sboeuf** [comment]
This PR is ready to be merged. The git submodules `runv` and `cc-runtime` will be removed in a subsequent PR.
The test PR https://github.com/kata-containers/runtime/pull/60 proves that both this PR #59 and #55 fix the CI for `kata-runtime`.
I think we want to move forward and have our CI fixed asap so that further PRs can really be gated by the CI, that's why I'll go ahead and merge it.

**Comment by WeiZhang555** [comment]
> It is barely reviewable given the patch size... Just some minor comments.

**Comment by sboeuf** [comment]
@WeiZhang555 Yes I agree, but this one was more like a huge copy/paste from `clearcontainers/runtime` into `cli`.

**Pull request merged**